src/sage/quadratic_forms/binary_qf.py

```diff
@@ -1636,6 +1678,15 @@ def solve_integer(self, n):
 
             return None
 
+        if algorithm == 'cornacchia':
+            if not (self._a.is_one() and self._b.is_zero() and self._c > 0):
+                raise ValueError("Cornacchia's algorithm requires a=1 and b=0 and c>0")
+            sol = pari.qfbcornacchia(self._c, n)
+            return tuple(map(ZZ, sol)) if sol else None
+
+        if algorithm != 'general':
+            raise NotImplementedError(f'algorithm {algorithm!r} is unknown')
```

Comment:
```suggestion
            raise ValueError(f'{algorithm!r} is not a valid algorithm')
```
This is consistent with other pars of Sage.